### DISCUSSION TITLE
wire:model field not changed after value change in updated()

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#2276] [1. Help] [answered]
Not sure if this is a bug or my misunderstanding.

I have a list where each item will have a quantity and price field. I'd like to update the total automatically. This should be straight forward, I thought. Our currency fields look like this:

```blade
<x-input.currency wire:model="items.{{ $key }}.subtotal" :symbol=$currency />
```

This is a custom blade component where we use AutoNumeric to format it as we like. I've removed all the formatting as it's largely coming from a Tailwind Component, but this is how it basically looks like:

```blade
@props([
    'symbol' => '€',
])

<div
    x-data="{ value: @entangle($attributes->wire('model')), autonumeric: undefined }"
    x-init="
        autonumeric = new AutoNumeric($refs.currency, value, {
            currencySymbol: '{{ $symbol }} ',
            decimalCharacter: ',',
            digitGroupSeparator: '.'
        });
    "
    class="flex rounded-md"
>
    <input
        x-ref="currency"
        x-on:blur="value = autonumeric.getNumber()"
        type="text"
        {{ $attributes->whereDoesntStartWith('wire:model') }}
    />
</div>
```

I've forwarded the full source of this while ago to Caleb, so it might make it in Surge.

Whenever I update the `quantity` or `price`, I'm using the `updated()` method from Livewire to set the `subtotal`:

```php
public function updated($name, $value)
{
    preg_match('/[0-9]/', $name, $matches);
    $key = $matches[0];

    $item = $this->items->get($key);
    $item['subtotal'] = 1234567;
    $this->items->put($key, $item);
}
```

Here's what I don't understand:

- The `subtotal` field isn't updated.
- I have added `@json($items)` to the blade view, this does return `1234567` as the `subtotal`.
- But it isn't part of the data being returned in the XHR request, it still says the `subtotal` is 0.

What am I missing? Is this a bug or should I do something else to make sure the value is updated?

**Comment by ju5t** [comment, marked as answer]
In the end all that I needed to add was `x-bind="value"` to the input of the component. Apparently AutoNumeric understands it should be formatted too, which is cool.

**Reply by jakjr** [reply]
Hello @ju5t ,

I am trying to use autoNumeric with livewire, but when the onBlur occurs the value of input get "blank".

Then, when onFocus occurs the value is displayed....

do you go through this?

thaks

**Reply by ju5t** [reply]
I have added `x-bind="value"` to the input. That's pretty much it.

**Reply by sooand** [reply]
@jakjr Did you find a solution to this? I'm having the same issue. Adding `x-bind="value"` doesn't help

**Reply by jakjr** [reply]
@sooand No.... there is some conflict between livewire (wire:model) and AutoNumetic when the onBlur occurs. I could not resolve... a try to change the autoNumeric config without success....

**Reply by ju5t** [reply]
Can you share a Laravel Playground setup where it does not work? I can have a quick look at it.

**Reply by sooand** [reply]
It seems adding "wire:ignore" helps. In the first div:
```
<div
    x-data="{ value: @entangle($attributes->wire('model')), autonumeric: undefined }"
    x-init="
        autonumeric = new AutoNumeric($refs.currency, value, {
            currencySymbol: '{{ $symbol }} ',
            decimalCharacter: ',',
            digitGroupSeparator: '.'
        });
    "
    class="flex rounded-md"
    wire:ignore
>
```
